ex2/tests/tests.py

```diff
@@ -1,55 +1,61 @@
+        for dep in dependencies:
+            content += F"{dep}\n"
+        if depth > 0:
+            content += "-r requirements1.txt\n"
+    else:
+        file_depth_level = int(filename[1:].split('.')[0][-1])
+        for dep in dependencies:
+            content += F"{file_depth_level}{dep}\n"
+        if file_depth_level < depth:
+            content += F"requirements{file_depth_level+1}.txt"
+    file_object = mock.mock_open(read_data=content).return_value
+    file_object.__iter__.return_value = content.splitlines(True)
+    return file_object
+
+
+def _open_circular_dep(filename: str, *args):
+    content = ""
+    _open_deep([], 1, "requirements.txt")
+    if "requirements1.txt" in filename:
+        content += "-r requirements.txt\n"
+    file_object = mock.mock_open(read_data=content).return_value
+    file_object.__iter__.return_value = content.splitlines(True)
+    return file_object
 
 
 class RequirementsProcessingTests(unittest.TestCase):
     def __init__(self, *args, **kwargs):
         self.basedir = ".\\requirements"
         super(RequirementsProcessingTests, self).__init__(*args, **kwargs)
 
-    def tearDown(self):
-        shutil.rmtree(self.basedir)
-
+    @mock.patch('ex2.requirements_processing.open', new=functools.partial(_open_deep, ["numpy>=1.0.0", "pandas<=1.0.0"], 0))
     def test_no_recursion(self):
-        create_mock_files(self.basedir, ["numpy>=1.0.0", "pandas<=1.0.0"])
         expected = "numpy>=1.0.0,pandas<=1.0.0"
         actual = RequirementsProcessor(self.basedir).requirements_file_to_string("requirements.txt")
         self.assertEqual(expected, actual)
 
+    @mock.patch('ex2.requirements_processing.open', new=functools.partial(_open_deep, ["numpy>=1.0.0", "pandas<=1.0.0"], 1))
+    @mock.patch('os.path.isfile', new=lambda x: True)
     def test_recursion(self):
-        create_mock_files(self.basedir, ["numpy>=1.0.0", "pandas<=1.0.0"], 1)
         expected = "numpy>=1.0.0,pandas<=1.0.0,1numpy>=1.0.0,1pandas<=1.0.0"
         actual = RequirementsProcessor(self.basedir).requirements_file_to_string("requirements.txt")
         self.assertEqual(expected, actual)
 
+    @mock.patch('ex2.requirements_processing.open', new=functools.partial(_open_circular_dep, "requirements1.txt"))
+    @mock.patch('os.path.isfile', new=lambda x: True)
     def test_circular_dep_throws_error(self):
-        create_mock_files(self.basedir, ["numpy>=1.0.0", "pandas<=1.0.0"], 1)
-        with open(os.path.join(self.basedir, "requirements1.txt"), "w") as f:
-            f.write("\n-r requirements.txt")
-        to_throw = functools.partial(RequirementsProcessor(self.basedir).requirements_file_to_string, "requirements.txt")
-        self.assertRaises(CircularDependencyError, to_throw)
+        to_throw = RequirementsProcessor(self.basedir).requirements_file_to_string
+        self.assertRaises(CircularDependencyError, to_throw, base_file='requirements.txt')
 
 
 if __name__ == '__main__':
```

Comment:
you miss some use cases. What if you get requirements.txt without -r? What if you get requirements.txt? 
What if you get -really requirements.txt? 
What if you get  "-r requirements.txt /n -r requirements.txt" in the same file?